src/AI/CoreLogic.pm

```diff
@@ -993,14 +993,14 @@ sub processStorageGet {
 sub processCartAdd {
 	if (AI::action eq "cartAdd" && timeOut(AI::args)) {
 		my $item = AI::args->{items}[0];
-		my $i = $item->{index};
-		my $invItem = $char->inventory->get($i);
+		my $index = $item->{index};
+		my $amount = $item->{amount};
+		my $invItem = $char->inventory->getByServerIndex($index);
```

Comment:
This might be more succinct:
```
my $invItem = $char->inventory->getByServerIndex($item->{index});
if ($invItem) {
    $messageSender->sendCartAdd($item->{index}, min($invItem->{amount}, $item->{amount} || $invItem->{amount}));
}
```